src/Symfony/Component/Notifier/Transport/AbstractTransport.php

```diff
@@ -74,7 +76,21 @@ public function send(MessageInterface $message): SentMessage
             $this->dispatcher->dispatch(new MessageEvent($message));
         }
 
-        return $this->doSend($message);
+        try {
+            $sentMessage = $this->doSend($message);
```

Comment:
I feel the code would be simpler if we merge the `if null !== dispatcher` together

```
if (null === $this->dispatcher) {
  return $this->doSend($message);
}

$this->dispatcher->dispatch(new MessageEvent($message));
try {
  $sentMessage = $this->doSend($message);
} catch (\Throwable $error) {
  $this->dispatcher->dispatch(new FailedMessageEvent($message, $
  throw $error;
}

$this->dispatcher->dispatch(new SentMessageEvent($message));

return $sentMessage;
```

same for nullTransport

Reply:
You're right, Thanks! I updated in this way